server.ts

```diff
@@ -0,0 +1,36 @@
+import * as express from 'express';
+import { TodoApp } from './src/todo';
+
+const todo = new TodoApp();
+
+const serv = express();
+serv.use(express.json());
+
+const port = 3000;
+
+serv.get('/', function(req, res, next) {
+    res.json(todo.listAllToDos());
+});
+
+serv.post('/', function(req, res, next) {
+    const data = req.body;
+    res.json(todo.addItem(data.description, data.dueAt));
+});
+
+serv.put('/:id/done', function(req, res, next) {
+    const id: string = req.params.id;
+    todo.checkItem(parseInt(id));
```

Comment:
This is great that you are using `parseInt` here.  A better thing would have been to check if you parsed int was an actual int or `NaN`.  If the latter you should throw a 400 since its a user error.  But the fact that you are parsing it and not relying on the user to do it right is a very good start.

Reply:
In general you _must never_ trust user input.  Just imagine a baddie manipulating it and then you can see the potential ways to abuse it.  there aren't many here but you can see why its important to sanitize and parse user input